pkg/security/podsecuritypolicy/provider.go

```diff
@@ -229,10 +229,28 @@ func (s *simpleProvider) ValidatePodSecurityContext(pod *api.Pod, fldPath *field
 			}
 
 			if fsType == extensions.HostPath {
-				if !psputil.AllowsHostVolumePath(s.psp, v.HostPath.Path) {
+				if len(s.psp.Spec.AllowedHostPaths) == 0 {
+					// If no allowed paths are specified then allow any path
+					// since host path volumes are allowed.
+					continue
+				}
+
+				allowedHostPath := psputil.GetAllowedHostPath(s.psp.Spec.AllowedHostPaths, v.HostPath.Path)
+				if allowedHostPath == nil {
 					allErrs = append(allErrs, field.Invalid(
 						field.NewPath("spec", "volumes").Index(i).Child("hostPath", "pathPrefix"), v.HostPath.Path,
 						fmt.Sprintf("is not allowed to be used")))
+					continue
+				}
+
+				if allowedHostPath.ReadOnly {
+					for cIdx, c := range pod.Spec.InitContainers {
```

Comment:
this isn't quite right. see the example at https://github.com/kubernetes/kubernetes/pull/58647/commits/107aabef06b766daecb71e7f16fa89094ec05fa2#r164038141

make sure there's a testcase for the overlapping-allowedHostPaths-with-different-readonly-settings scenario

Reply:
That link isn't going any where specific for me :/

Reply:
That link isn't going any where specific for me.

Reply:
weird, was just a link to the comment above

Reply:
still outstanding

Reply:
Mind looking at the tests I added on the 10th? Is it still outstanding?  I thought I covered this there.